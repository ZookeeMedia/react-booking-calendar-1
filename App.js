// @flow

import React from 'react';
import { Provider } from 'react-redux';
import { TabNavigator, StackNavigator } from 'react-navigation';
import { StyleSheet, Text, View } from 'react-native';
import store from './store';
import Login from './screens/auth/Login/';
import Signup from './screens/auth/Signup/';
import AvailabilityCalendar from './screens/availability/AvailabilityCalendar/';
import SelectTimes from './screens/availability/SelectTimes/';
import BookingsOverviewUser from './screens/bookings/BookingsOverviewUser/';

export default class App extends React.Component {
  render() {
    const MainNavigator = new TabNavigator({
      auth: { screen: new StackNavigator({
        login: { screen: Login },
        signup: { screen: Signup },
      }) },
      main: { screen: new TabNavigator({
        availability: { screen: new StackNavigator({
          availabilityCalendar: { screen: AvailabilityCalendar },
          selectTimes: { screen: SelectTimes },
        }) },
        bookings: { screen: new StackNavigator({
          bookingsUser: { screen: BookingsOverviewUser }
        }) }
      }, { tabBarOptions: { showLabel: false } }),

    },
    }, { navigationOptions: { tabBarVisible: false }
  })

    return (
      <Provider store={store}>
        <View style={styles.container}>
          <MainNavigator />
        </View>
      </Provider>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff'
  },
});
