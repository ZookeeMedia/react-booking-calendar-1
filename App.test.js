// @flow

import 'react-native';
import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';
import App from './App';

import renderer from 'react-test-renderer';

test('renders correctly jest react test renderer', () => {
  const tree = renderer.create(
    <App />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renderd correctly with toJSON', () => {
  const component = shallow(<App />);
  const tree = toJSON(component);
  expect(tree).toMatchSnapshot;
});

test('renders as expected', () => {
  const component = shallow(<App />)
  expect(component.contains('Open up App.js to start working on your app')).toBe(true);
});
