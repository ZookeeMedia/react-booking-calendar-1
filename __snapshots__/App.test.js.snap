// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly jest react test renderer 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    Open up App.js to start working on your app
  </Text>
</View>
`;
